patterns/ecs-v1/aws

```diff
@@ -1,14 +1,22 @@
-S3_REQUEST_LINE (?:%{WORD:verb} %{NOTSPACE:request}(?: HTTP/%{NUMBER:httpversion})?|%{DATA:rawrequest})
+# TODO TEMP: revert URIQUERY
+URIQUERY [A-Za-z0-9$.+!*'|(){},~@#%&/=:;_?\-\[\]<>]*
 
-S3_ACCESS_LOG %{WORD:owner} %{NOTSPACE:bucket} \[%{HTTPDATE:timestamp}\] %{IP:clientip} %{NOTSPACE:requester} %{NOTSPACE:request_id} %{NOTSPACE:operation} %{NOTSPACE:key} (?:"%{S3_REQUEST_LINE}"|-) (?:%{INT:response:int}|-) (?:-|%{NOTSPACE:error_code}) (?:%{INT:bytes:int}|-) (?:%{INT:object_size:int}|-) (?:%{INT:request_time_ms:int}|-) (?:%{INT:turnaround_time_ms:int}|-) (?:%{QS:referrer}|-) (?:"?%{QS:agent}"?|-) (?:-|%{NOTSPACE:version_id})
+S3_REQUEST_LINE (?:%{WORD:[http][request][method]} %{NOTSPACE:[url][original]}(?: HTTP/%{NUMBER:[http][version]})?)
 
-ELB_URIPATHPARAM %{URIPATH:path}(?:%{URIPARAM:params})?
+S3_ACCESS_LOG %{WORD:[aws][s3access][bucket_owner]} %{NOTSPACE:[aws][s3access][bucket]} \[%{HTTPDATE:timestamp}\] (?:-|%{IP:[client][ip]}) (?:-|%{NOTSPACE:[client][user][id]}) %{NOTSPACE:[aws][s3access][request_id]} %{NOTSPACE:[aws][s3access][operation]} (?:-|%{NOTSPACE:[aws][s3access][key]}) (?:-|"%{S3_REQUEST_LINE:[aws][s3access][request_uri]}") (?:-|%{INT:[http][response][status_code]:int}) (?:-|%{NOTSPACE:[aws][s3access][error_code]}) (?:-|%{INT:[aws][s3access][bytes_sent]:int}) (?:-|%{INT:[aws][s3access][object_size]:int}) (?:-|%{INT:[event][duration]:int}) (?:-|%{INT:[aws][s3access][turn_around_time]:int}) "(?:-|%{DATA:[http][request][referrer]})" "(?:-|%{DATA:[user_agent][original]})" (?:-|%{NOTSPACE:[aws][s3access][version_id]})(?: (?:-|%{NOTSPACE:[aws][s3access][host_id]}) (?:-|%{NOTSPACE:[aws][s3access][signature_version]}) (?:-|%{NOTSPACE:[aws][s3access][cipher_suite]}) (?:-|%{NOTSPACE:[aws][s3access][authentication_type]}) (?:-|%{NOTSPACE:[aws][s3access][host_header]}) (?:-|%{NOTSPACE:[aws][s3access][tls_version]}))?
```

Comment:
Is the value you're putting in `event.duration` nanoseconds? If not, probably better to capture this in a custom field.

Perhaps you could use the [TLS fields](https://www.elastic.co/guide/en/ecs/current/ecs-tls.html) as well towards the end?

* `[aws][s3access][cipher_suite]` => `[tls][cipher]`
* `[aws][s3access][tls_version]` => `[tls][version]` & `[tls][version_protocol]`

Reply:
> Is the value you're putting in event.duration nanoseconds? If not, probably better to capture this in a custom field.

it isn't was blindly following Beats which copies it's [`aws.s3access.total_time`](https://github.com/elastic/beats/blob/v7.9.2/x-pack/filebeat/module/aws/s3access/test/s3_server_access.log-expected.json#L19) to [`event.duration`](https://github.com/elastic/beats/blob/v7.9.2/x-pack/filebeat/module/aws/s3access/test/s3_server_access.log-expected.json#L27) (which is millis). 
thanks for the catch, should have read the ECS docs!

> Perhaps you could use the TLS fields as well towards the end?

yeah we could definitely do the `[tls][cipher]` by renaming the field (also captured in `CLOUDFRONT_ACCESS_LOG`),

however we can not do `[tls][version_protocol]` since ECS prescribes the field in lowercase and we would match uppercase -> best we could do is having half baked: 

`[aws][s3access][tls_version] = 'TLSv1.2'` and `[tls][version] = '1.2'` with a composite capture, the reason I haven't rushed into that is that if felt half done -> user still needs to extract `[tls][version_protocol]` so he could just as well be okay to only start with `[aws][s3access][tls_version]`